### PR TITLE
Exit behaviour

### LINKED ISSUE #3: Improve exit behaviour
Currently fan2go doesn't do anything when exiting. This can lead to situations where the fans are "stuck" on a very low PWM level, which is a problem when the load increases without fan2go or a different fan control system running.

# Solution
fan2go should try to restore the `pwm_enabled` value of configured fans to their original value when exiting.  
If this is not working for some reason, or the value was `1` (meaning manual control) fan2go should try to set all configured fans to manual and maximum speed (255).

### PR DESCRIPTION
**Pull request by markusressel**
remember original pwm_enabled value per fan
added signal handler to restore fan configurations when exiting
fixed inverted error comparison
removed unused code

fixes #3

### CODE REVIEW

File: internal/backend.go [hunk 1]
```diff
@@ -72,6 +76,36 @@ func Run(verbose bool) {
...
+							continue
+						}
+					}
+					err := setPwm(fan, MaxPwmValue)
+					if err != nil {
+						log.Printf("WARNING: Unable to revert fan %s, make sure it is running!", fan.Config.Id)
+					}
+				}
+			}
+		}
```
Comment (metalmatze) [inline internal/backend.go:103]: I'd rather put this into a separate function and the rest of this setupExitHandler straight into `func main()`.
Usually we try to not hide goroutines in function calls either. Make the caller create the goroutine, in this case the `func main()`.

File: internal/backend.go [hunk 2]
```diff
@@ -72,6 +76,36 @@ func Run(verbose bool) {
...
+		log.Println("Exiting...")
+		os.Exit(0)
```
Comment (metalmatze) [inline internal/backend.go:105]: This is simply doing explicitly what Go already does implicitly. `os.Exit(0)` is the default.

**Review by metalmatze — CHANGES_REQUESTED** [review]
Overall this can probably benefit from using https://github.com/oklog/run
Then you'd even be able to use https://github.com/oklog/run/blob/master/actors.go#L13 for handling the signals.
Nevertheless this was a good exercise! :blush:

File: internal/backend.go (hunk 2 above)
Comment (markusressel) [reply internal/backend.go:105]: Right, but when is it doing that? I put it there to force it after the signal handler is done. Will go call `os.Exit` right after the signal handler returns?

**Pull request merged**